### PR TITLE
refactor: Remove effects/reducers from cart public API

### LINKED ISSUE #4507: Remove effects/reducers from cart store public api
Effects and reducers from cart are exported in public api.

Remove `export * from './store/index';`
Add export for actions instead.

### PR DESCRIPTION
**Pull request by marlass**
Closes #4507

### CODE REVIEW

File: projects/schematics/src/migrations/2_0/removed-public-api.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+          continue;
+        }
+        // 'source' has to be reloaded after each committed change
+        const source = getTsSourceFile(tree, sourcePath);
+
+        const changes = insertCommentAboveIdentifier(
+          sourcePath,
+          source,
+          removedNode.node,
+          buildComment(removedNode.comment ?? `'${removedNode.node}' is no longer part of the public API. Please look into migration guide for more information`)
```
Comment (znikola) [inline projects/schematics/src/migrations/2_0/removed-public-api.ts]: Are you referring to `docs/migration/2_0.md`?
Maybe we can think about printing the link either to the customer's console, or as part of every comment (which will probably bring a lot of noise). Take a look at this comment for how the link will look like: https://github.com/SAP/cloud-commerce-spartacus-storefront/issues/6588#issuecomment-593862579

Comment (marlass) [reply projects/schematics/src/migrations/2_0/removed-public-api.ts]: As discussed:
- links in each comment brings to much noise
- better to add link to docs as final schematics output

File: projects/schematics/src/shared/utils/file-utils.ts [hunk 2]
```diff
@@ -732,6 +738,95 @@ export function insertCommentAboveIdentifier(
...
 }
 
+export function getImportDeclarations(
+  source: ts.SourceFile,
+  importPath: string
+): ts.ImportDeclaration[] {
+  const imports = getSourceNodes(source).filter(
+    node => node.kind === ts.SyntaxKind.ImportDeclaration
+  );
+  if (!imports) {
```
Comment (znikola) [inline projects/schematics/src/shared/utils/file-utils.ts]: I believe `imports` will always be defined, right?

File: projects/schematics/src/shared/utils/file-utils.ts [hunk 3]
```diff
@@ -732,6 +738,95 @@ export function insertCommentAboveIdentifier(
   return changes;
 }
 
+export function getImportDeclarations(
```
Comment (znikola) [inline projects/schematics/src/shared/utils/file-utils.ts]: Do we need this exported?
Also, could we somehow merge logic from `getImportDeclarationNode` and this method?

File: projects/schematics/src/shared/utils/file-utils.ts [hunk 4]
```diff
@@ -732,6 +738,95 @@ export function insertCommentAboveIdentifier(
...
+    return [];
+  }
+  return imports.filter(
+    imp =>
+      ((imp as ts.ImportDeclaration).moduleSpecifier as ts.StringLiteral)
+        .text === importPath
+  ) as ts.ImportDeclaration[];
+}
+
+export function filterNamespacedImports(
```
Comment (znikola) [inline projects/schematics/src/shared/utils/file-utils.ts]: Do we need this exported?

File: projects/schematics/src/shared/utils/file-utils.ts [hunk 5]
```diff
@@ -732,6 +738,95 @@ export function insertCommentAboveIdentifier(
...
+  imports: ts.ImportDeclaration[]
+): ts.ImportDeclaration[] {
+  return imports
+    .filter(imp => (imp.importClause?.namedBindings as any)?.name)
+    .filter(Boolean);
+}
+
+export function filterNamedImports(
```
Comment (znikola) [inline projects/schematics/src/shared/utils/file-utils.ts]: Do we need this exported?

File: projects/schematics/src/shared/utils/file-utils.ts (hunk 2 above)
Comment (marlass) [reply projects/schematics/src/shared/utils/file-utils.ts]: Yep

File: projects/schematics/src/shared/utils/file-utils.ts (hunk 3 above)
Comment (marlass) [reply projects/schematics/src/shared/utils/file-utils.ts]: Fixed. Used this method in `getImportDeclarationNode`.

**Review by Platonn — APPROVED** [review]
Removals from cart's public API look good to me.

I'm wondering only if we could have consistent properties in the helpers:
- importPath
- node vs. className.

@znikola What do you think about changin `className` to `node` in the constructor schematics helper - for consistency?

**Comment by kacperknapik** [comment]
**QA results:**
removed entities not present in public API

**Pull request merged**